src/Layout/Presentation/SingleParty/SingleParty.tsx

```diff
@@ -0,0 +1,257 @@
+import * as React from "react";
+import ReactTable from "react-table";
+import { DistrictResult, AlgorithmType } from "../../../computation/computation-models";
+import { toSum } from "../../../utilities/reduce";
+import { DisproportionalityIndex } from "../presentation-models";
+import { checkExhaustively } from "../../../utilities";
+import { PartySelect } from "./PartySelect";
+import { norwegian } from "../../../utilities/rt";
+import { roundNumber } from "../../../utilities/number";
+import {
+    getVotesToVulnerableSeatMap,
+    getQuotientsToVulnerableSeatMap,
+    getVulnerableSeatByQuotient,
+    getVulnerableSeatByVotes,
+    createPartyResultMap,
+    VulnerableVotes,
+} from "../../../utilities/district";
+import { isQuotientAlgorithm } from "../../../computation/logic";
+
+export interface SinglePartyProps {
+    districtResults: DistrictResult[];
+    partySelected: string;
+    selectParty: (event: React.ChangeEvent<HTMLSelectElement>) => void;
+    decimals: number;
+    disproportionalityIndex: DisproportionalityIndex;
+    algorithm: AlgorithmType;
+    districtThreshold: number;
+    districtSeats: number;
+    firstDivisor: number;
+    partyMap: _.Dictionary<string>;
+}
+
+interface SinglePartyResult {
+    district: string;
+    votes: number;
+    percentVotes: number;
+    districtSeats: number;
+    levelingSeats: number;
+    totalSeats: number;
+    marginInVotes: number;
+    seatWinner: boolean;
+    closestVotes: boolean;
+    quotientMargin: number;
+    closestQuotient: boolean;
+    proportionality: number;
+}
+
+function getMarginInVotes(partyCode: string, vulnerableMap: Map<string, number> | undefined, vulnerableVotes: VulnerableVotes | undefined, firstDivisor: number): number {
+    if (vulnerableMap) {
+        const vulnerableVotes = vulnerableMap.get(partyCode);
+        if (vulnerableVotes) {
+            return vulnerableVotes;
+        }
+    }
+
+    if (vulnerableVotes) {
+        return Math.floor(vulnerableVotes.winner.quotient * firstDivisor) + 1;
+    }
+
+    return 0;
+}
+
+function createPartyCodeList(districtResults: DistrictResult[]): string[] {
+    const partyCodeSet = new Set<string>();
+    
+    for (let i = 0; i < districtResults.length; i++) {
```

Comment:
Codacy found an issue: [Expected a 'for-of' loop instead of a 'for' loop with this simple iteration](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=6089581)